var/spack/repos/builtin/packages/neurodamus-core/package.py

```diff
@@ -42,17 +46,44 @@ def install(self, spec, prefix):
         shutil.copytree('mod', prefix.mod)
         if spec.satisfies('+python'):
             copy_tree('resources/neurodamus-py', prefix.python)
-        if spec.satisfies('+common'):
-            copy_all('resources/common/hoc', prefix.hoc)
-            copy_all('resources/common/mod', prefix.mod)
 
         filter_file(r'UNKNOWN_CORE_VERSION', r'%s' % spec.version, prefix.hoc.join('defvar.hoc'))
-
         try:
             commit_hash = self.fetcher[0].get_commit()
             filter_file(r'UNKNOWN_CORE_HASH', r'\'%s\'' % commit_hash, prefix.hoc.join('defvar.hoc'))
         except Exception as e:
-            tty.warn(e)
+            tty.warn(str(e))
+
+        # +Common will bring common mods and build a bare nrnmechlib
+        if spec.satisfies('+common'):
+            copy_all('resources/common/hoc', prefix.hoc)
+            copy_all('resources/common/mod', prefix.mod)
+
+            with working_dir(prefix):
+                which('nrnivmodl')('-incflags', '-DDISABLE_REPORTINGLIB', 'mod')
+
+                bindir = os.path.basename(self.neuron_archdir)
+                special = join_path(bindir, 'special')
+                open('.bindir', 'w').write(bindir)
+
+                if not os.path.isfile(special):
+                    raise Exception("Failed to build neurodamus core mods")
+
+                # Cleanup
+                for f in find(bindir, '*.lo'): os.remove(f)
+                for f in find(bindir, '*.mod'): os.remove(f)
+                for f in find(bindir + '/.libs', '*.o'): os.remove(f)
+                os.mkdir(bindir + "/mod2c")
+                for f in find(bindir, "*.c*"): shutil.move(f, bindir + "/mod2c/")
 
     def setup_environment(self, spack_env, run_env):
         run_env.set('HOC_LIBRARY_PATH', self.prefix.hoc)
+        if self.spec.satisfies('+common'):
+            run_env.set('MOD_LIBRARY_PATH', self.prefix.mod)
+            bindir_info = self.prefix.join('.bindir')
+            if os.path.isfile(bindir_info):
+                bindir = open(bindir_info, 'r').readline()
+                mechlib = find_libraries('libnrnmech', join_path(self.prefix, bindir, '.libs'))[0]
+                run_env.set('NRNMECH_LIB_PATH', mechlib)
```

Comment:
`sim-model` also set `NRNMECH_LIB_PATH`. I wonder if we should do `prepend` instead in both cases?

Reply:
You raised a good point. 
 - If we don't depend on neurodamus-core+model (I think we shouldn't!) then it doesn't matter
 - If some neurodamus depends on it, current recipes build "full" nrnmech, and therefore the neurodamus-core NRNMECH_LIB_PATH should be ignored

In the second case, do we have any guarantee that neurodamus-model module is happening after neurodamus-core?

Reply:
But what if user load multiple modules? For example, neurodamus-hippocampus followed by neurodamus-core OR vice-a-versa?

Reply:
Maybe we should mark neurodamus-core and neurodamus-xxx incompatible... it doesn't make sense to load two simultaneously

Reply:
Ok. as discussed, lets got with this for now.